src/loaders/lottie/tvgLottieAnimation.cpp

```diff
@@ -54,3 +54,20 @@ unique_ptr<LottieAnimation> LottieAnimation::gen() noexcept
 {
     return unique_ptr<LottieAnimation>(new LottieAnimation);
 }
+
+bool LottieAnimation::segment(const char* marker)
+{
+    auto loader = pImpl->picture->pImpl->loader;
+    if (!loader || !loader->animatable()) return false;
+    
+    if (!marker) {
+        static_cast<FrameModule*>(loader)->segment(0.0, 1.0);
```

Comment:
It might be necessary to cast directly to LottieLoader rather than to FrameModule.